src/dtype.rs

```diff
@@ -217,74 +273,95 @@ impl DataType {
 /// });
 /// ```
 pub unsafe trait Element: Clone + Send {
-    /// `DataType` corresponding to this type.
-    const DATA_TYPE: DataType;
-
-    /// Returns if the give `dtype` is convertible to `Self` in Rust.
-    fn is_same_type(dtype: &PyArrayDescr) -> bool;
-
-    /// Returns the corresponding
-    /// [Enumerated Type](https://numpy.org/doc/stable/reference/c-api/dtype.html#enumerated-types).
-    #[inline]
-    fn npy_type() -> NPY_TYPES {
-        Self::DATA_TYPE.into_ctype()
-    }
+    /// Flag that indicates whether this type is 'plain old data'.
+    ///
+    /// It should be set to true for all trivially copyable types (like scalar types
+    /// and record/array types only containing trivially copyable fields and elements).
+    ///
+    /// This flag should *always* be set to `true` for object types.
+    const IS_POD: bool;
```

Comment:
I like `IS_COPYABLE` or so

Reply:
Here's a few arguments if I might:
- `IS_COPYABLE` is a bit confusing: you can have `#[derive(Copy)] struct Foo {}` type in Rust which you stick into a `Py<Foo>` numpy array that you can `.copy()` in Python.
- It's *not* only about copying. E.g. it may matter when extracting fields out of record arrays, extracting subarrays etc.

Alternative option is `HAS_OBJECT`, I guess...

Reply:
Yes, I agree. We should stay closer to Rust's `Copy` terminology here. POD might be familiar to C++ users, but it is rarely used in Rust code.

Reply:
The argument on `struct Foo` applies just as well to POD, as `Foo` is POD, but `Py<Foo>` is not. The point is also not to limit the discussion to copying, but to align the nomenclature to Rust's `Copy` trait (on the element type, which would be `Py<Foo>`, not `Foo`).

Reply:
If we want to keep `IS_POD` / `HAS_OBJECT` around, we could also sprinkle some `debug_assert`s around, at least to check the trivial 'O' dtype case.

Reply:
You mean `IS_COPY` ;-) But yeah, adding `debug_assert`s checking that this matches the descriptor is certainly a good idea.

Reply:
Ok, I've renamed it to `IS_COPY` and extended the docstring a bit to also mention record types.